src/app/_mermaid_common/forms/directives/validators/futureyear.dir.js

```diff
@@ -0,0 +1,31 @@
+angular.module('mermaid.forms').directive('futureyear', function() {
+  'use strict';
+  return {
+    require: 'ngModel',
+    link: function(scope, elem, attr, ngModel) {
+      var today = new Date().getTime();
+      ngModel.validator_messages = ngModel.validator_messages || {};
+      ngModel.validator_messages[
+        window.appConfig.validatorPrefix + 'futureyear'
+      ] = 'Date is in future';
+
+      function validate() {
+        var is_valid = true;
+        var dateInput = new Date(ngModel.$modelValue).getTime();
+        var dateDifference = Math.floor((today - dateInput) / 86400000);
```

Comment:
Do we need some kind of leeway here to avoid false positives? As I read the code, this will trigger if the entered time is > 12 hours in the future from now, is that right? If so then that seems good.